lib/util/s3_setup.js

```diff
@@ -24,3 +26,128 @@ module.exports.detect = function(to, config) {
     }
   }
 };
+
+module.exports.get_s3 = function(config) {
+
+  // if not mocking then setup real s3.
+  if (!process.env.node_pre_gyp_mock_s3) {
```

Comment:
I would prefer having this be `if (process.env.node_pre_gyp_mock_s3) {` so that it is an early return if the custom environment setting is defined. The reason is so that the default is to have the application use real s3, unless this custom env is set.

Reply:
changing. i did it the way i did so the "normal" return came first, but both approaches make sense to me - half full/half empty - just depends.